istio/datadog_checks/istio/legacy_1_4.py

```diff
@@ -20,6 +20,7 @@
 
 class LegacyIstioCheck_1_4(OpenMetricsBaseCheck):
 
+    DEFAULT_METRIC_LIMIT = 3000
```

Comment:
Why?

Reply:
The legacy check supports enabling multiple endpoints for one check and the metrics collection is a little under 2k. With the option enabled for the mesh endpoint, that default 2000 limit is reached and some metrics are skipped.

Reply:
We remove the limit entirely for official integrations now. Let's do: https://github.com/DataDog/integrations-core/blob/22f24a30f05c29d5eb9de9d20ef3ef2f878f79cd/amazon_msk/datadog_checks/amazon_msk/amazon_msk.py#L13

Reply:
But the OpenmetricsBaseCheck seems to default at 2k. Should we update that to `0`?

https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/checks/openmetrics/base_check.py#L35